401-advanced-javascript-billybunn/lab-07-testing
# Submission - testing

### Testing Practice
- [x] Create a repository called `lab-07-testing` and copy the contents of the `starter-code/testing` folder into it.
- [ ] Complete the following tasks
  * Write unit tests for the middleware
    * Mock `request`, `response`, `next`
    * Make assertions on their use/modifications
    * Spy on next
  * Start the server and visually verify your routes
    * /a should work
    * /e should error
    * anything else should 404
  * Stop the server and complete tests for it using `supertest`
    * Make note of how things are wired and provide answers to the embedded questions
